process-document/src/test/kotlin/com/ritense/processdocument/service/DocumentDelegateServiceTest.kt

```diff
@@ -0,0 +1,316 @@
+import org.junit.jupiter.api.Assertions.assertEquals
+import org.junit.jupiter.api.BeforeEach
+import org.junit.jupiter.api.Test
+import org.mockito.ArgumentMatchers.any
+import org.mockito.kotlin.mock
+import org.mockito.kotlin.verify
+import org.mockito.kotlin.whenever
+import java.time.LocalDateTime
+import java.util.UUID
+import java.util.Optional
+
+
+internal class DocumentDelegateServiceTest : BaseTest() {
+
+    lateinit var processDocumentService: ProcessDocumentService
+    lateinit var documentService: DocumentService
+    lateinit var jsonSchemaDocumentService: JsonSchemaDocumentService
+    lateinit var documentDelegateService: DocumentDelegateService
+
+    lateinit var definition: JsonSchemaDocumentDefinition
+    lateinit var delegateExecutionFake: DelegateExecution
+
+    private val STREET_NAME = "street"
+    private val HOUSE_NUMBER = "3"
+    private val NO = false
+
+    private val documentId = "11111111-1111-1111-1111-111111111111"
+    private val processInstanceId = "00000000-0000-0000-0000-000000000000"
+
+    @BeforeEach
+    fun setup() {
+        definition = definition()
+        documentSequenceGeneratorService = mock()
+        whenever(documentSequenceGeneratorService.next(any())).thenReturn(1L)
+        processDocumentService = mock()
+        documentService = mock()
+        jsonSchemaDocumentService = mock()
+        documentDelegateService = DocumentDelegateService(
+            processDocumentService,
+            documentService,
+            jsonSchemaDocumentService
+        )
+        delegateExecutionFake =
+            DelegateExecutionFake("id").withProcessBusinessKey("56f29315-c581-4c26-9b70-8bc818e8c86e");
+    }
+
+    @Test
+    fun `get modifiedOn from document`() {
+        val documentMock = mock<JsonSchemaDocument>()
+        val delegateExecutionFake = DelegateExecutionFake("id")
+            .withProcessInstanceId(processInstanceId)
+
+        val jsonSchemaDocumentId = JsonSchemaDocumentId.existingId(UUID.fromString(documentId))
+        val modifiedOn = LocalDateTime.now()
+
+        whenever(documentMock.modifiedOn()).thenReturn(Optional.of(modifiedOn))
+        whenever(
+            processDocumentService.getDocumentId(
+                CamundaProcessInstanceId(processInstanceId),
+                delegateExecutionFake
+            )
+        )
+            .thenReturn(jsonSchemaDocumentId)
+
+        whenever(jsonSchemaDocumentService.getDocumentBy(jsonSchemaDocumentId))
+            .thenReturn(documentMock)
+
+        val modifiedOnResult = documentDelegateService.getDocumentModifiedOn(delegateExecutionFake)
+
+        assertEquals(modifiedOnResult, modifiedOn)
+
+        verify(processDocumentService).getDocumentId(CamundaProcessInstanceId(processInstanceId), delegateExecutionFake)
+        verify(jsonSchemaDocumentService).getDocumentBy(jsonSchemaDocumentId)
+    }
+
+    @Test
+    fun `get assigneeId from document`() {
```

Comment:
These tests are 95% the same. Can you change them so that there's not as much code duplication?

Reply:
I will have a look. However code duplication in test code isn't as bad as in source code. If a test fails it should be clear (readable) to see what goes wrong. In this case I agree '95%' is a bit to much.